### PR TITLE
fix: added condition to disable free text entry

### LINKED ISSUE #518: Add option to disable free text entry in MultiSelect
Currently users can enter free text into MultiSelect and it becomes a value. Sometimes you just want to limit users to the available values.

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Added condition to disable free text entry in multiselect

## Issue ticket number and link

Closes #518 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ export type MultiSelectProps = {
   items?: Expression<SelectOption[]>;
   onItemSelect?: EnsembleAction;
   hintStyle?: EnsembleWidgetStyles;
+  mode?: "multiple" | "tags";
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:43]: Pick a better attribute name - it needs to be intuitive to the end user not just a passthrough of whatever internal component you are using. That is a poor abstraction.

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 2]
```diff
@@ -127,12 +128,11 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
...
-      values?.options.some((option) =>
-        option.label.toString().toLowerCase().startsWith(value.toLowerCase()),
-      )
-    )
-      setNewOption("");
+    const isOptionExist = values?.options.some((option) =>
+      option.label.toString().toLowerCase().startsWith(value.toLowerCase()),
+    );
+
+    if (isOptionExist || values?.mode !== "tags") setNewOption("");
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:135]: Why is this required if the antd mode already handles the new option?

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/widgets/Form/MultiSelect.tsx (hunk 2 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/Form/MultiSelect.tsx:135]: To remove this bottom message when we disable the tags mode, we need this condition

image

File: apps/kitchen-sink/src/ensemble/screens/forms.yaml [hunk 3]
```diff
@@ -99,6 +99,7 @@ View:
                         labelKey: firstName
                         valueKey: email
                         items: ${ensemble.storage.get('dummyData')}
+                        mode: "multiple"
```
Comment (evshi) [inline apps/kitchen-sink/src/ensemble/screens/forms.yaml:102]: This needs to be updated.

**Review by evshi — APPROVED** [review]

**Pull request merged**